crates/sui-framework/packages/deepbook/sources/math.move

```diff
@@ -6,25 +6,123 @@ module deepbook::math {
     const FLOAT_SCALING: u64 = 1_000_000_000;
     const FLOAT_SCALING_U128: u128 = 1_000_000_000;
 
-    public fun mul(x: u64, y: u64): u64 {
-        let x = (x as u128);
-        let y = (y as u128);
-        ((x * y / FLOAT_SCALING_U128) as u64)
+    // <<<<<<<<<<<<<<<<<<<<<<<< Error codes <<<<<<<<<<<<<<<<<<<<<<<<
+    const EUnderflow: u64 = 1;
+    // <<<<<<<<<<<<<<<<<<<<<<<< Error codes <<<<<<<<<<<<<<<<<<<<<<<<
+
+    // multiply two floating numbers
+    public fun unsafe_mul(x: u64, y: u64): u64 {
```

Comment:
```suggestion
    public(friend) fun unsafe_mul(x: u64, y: u64): u64 {
```

Reply:
This will give us flexibility to change in the future

Reply:
Here, if we change the modifier from public to public(friend), it will lead to more restrictions on the math module, and the math module does not have any behavior to modify the state, so  can you give some explanations why this can give us more flexibiluity to change?